### DISCUSSION TITLE
Is there an API for adding secrets to repositories?

### DISCUSSION THREAD

**Discussion by DaHuoKolmostar** [community/community#25032] [Actions]
<p>Is there an API for adding secrets to repositories? I lookup up the <a href="https://developer.github.com/v3/" rel="nofollow noopener">REST API v3</a>, but found nothing.</p>
<p>Is there already one? Or is there going to be one?</p>
<p>Cheers!</p>

**Comment by andymckay** [comment, marked as answer]
<p>There currently isn’t an API for adding secrets. We are working on it though and should have it available in the new year.</p>

**Comment by mpdude** [comment]
<p>The API has been made available as beta this week.</p>
<p>Also, we have written a tool to help with managing secrets across a range of repos: <a href="https://github.com/webfactory/secret-spreader" rel="nofollow noopener">https://github.com/webfactory/secret-spreader</a>. See the README for full details.</p>
<p>Feedback is very much appreciated!</p>
